### PR TITLE
Support empty string as a commit message when merging a pull request

### PR DESCRIPTION
**Pull request by john-m-liu**
Fixes https://github.com/google/go-github/issues/1815

- changes the CommitMessage field in the request to merge a PR to a pointer
- adds a new boolean flag to the PullRequestOptions struct, which will make the commit message null if set and the given message is ""

### CODE REVIEW

**Comment by john-m-liu** [comment]
@googlebot I signed it!

File: github/pulls_test.go
Comment (gmlewis) [inline github/pulls_test.go:821]: According to #1815, this will have no commit message, which is the case that I thought you added the `DontDefaultIfBlank` flag for.

File: github/pulls_test.go
Comment (gmlewis) [inline github/pulls_test.go:831]: According to #1815, this will use a default commit message, which I don't think is what you want when using the new `DontDefaultIfBlank` flag.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @john-m-liu , but I don't think this is working the way you want it to.  See comments below.

To solve the problem, I believe you may need to use an additional helper struct that simply removes the `omitempty` JSON tag for the `CommitMessage` when using the new flag.

Note that you need to keep the original one and then switch over to the new one internally for the new special case with the flag and the empty commit message.

File: github/pulls.go
Comment (john-m-liu) [reply github/pulls.go:434]: fixed

File: github/pulls_test.go
Comment (john-m-liu) [reply github/pulls_test.go:821]: doh, I inverted the boolean 🤦 fixed this in the function body, which I felt was preferable to creating a duplicate struct

File: github/pulls_test.go
Comment (john-m-liu) [reply github/pulls_test.go:831]: fixed

File: github/pulls.go
Comment (gmlewis) [inline github/pulls.go:461]: There's no need for a new variable here since we know at this point that `commitMessage == ""`.

Comment (john-m-liu) [reply github/pulls.go:461]: fixed

**Review by gmlewis — APPROVED** [review]
Thank you, @john-m-liu !
LGTM.

Awaiting second LGTM before merging.

**Review by wesleimp — APPROVED** [review]
💙

**Comment by gmlewis** [comment]
Thank you, @wesleimp !
Merging.

**Pull request merged**
